app/forms/refunds/confirmation_form.rb

```diff
@@ -0,0 +1,11 @@
+module Refunds
+  class ConfirmationForm < Form
+    attribute :application_reference, String
+    attribute :submitted_at, Date
+
+    def initialize(resource, &block)
+      raise 'Session has no persisted refund' unless resource.respond_to?('_refund_id')
```

Comment:
It shouldn't happen - it really is an exception, not a user error.  I simply put this in there to catch the error early if it did happen rather than await the confusion when it tries saving with no input data.
There is already a before action in the refunds controller that looks for missing input data (sessions table) and if its not present, the process is restarted with a flash message to the user

Reply:
OK, thanks for the info :)